### DISCUSSION TITLE
Kick/Ban all clients with the same IP

### DISCUSSION THREAD

**Discussion by iBoonie** [J-Tanzanite/Little-Anti-Cheat#113] [Ideas]
In my servers we have come across where a bunch of bots join the server. In my local fork I have it so if a cheater is detected, it will check all connected clients for the same IP and kick/ban them. Would this be something you would want added, I can submit a pull request easy enough.

**Comment by azalty** [comment]
If added, it should probably be opt-in and toggleable

Didn't know botting was a thing though. I don't really see the point too.

**Comment by iBoonie** [comment]
Yeah I agree it should be toggleable.
Its a problem in TF2, not sure in other games though. In my 32 player servers sometimes (rarely) a wave of 10+ would join all from the same IP.

**Comment by J-Tanzanite** [comment]
> Would this be something you would want added

>Yeah I agree it should be toggleable.
Its a problem in TF2, not sure in other games though. In my 32 player servers sometimes (rarely) a wave of 10+ would join all from the same IP.

I think this is niche enough to the point where it shouldn't be in Lilac?
I dunno, it affects one game (afaik), and I don't think most server owners would want this?
This idea of yours is not bad btw, this catbot issue has been a problem for a long time now, and I have at points considered tackling it.
I'm just not sure this is the right solution, and if other server owners would use this.

One thing I have considered in the past, is adding an IP checker when people join the server.
An optional (Disabled by default) misc feature that limits how many players can have the same IP.
For instance, setting the limit to 2, would mean that only two players can have the same IP, if a third player tries to connect from the same IP, they get rejected (kicked / cannot join).
This player IP limit approach, seems like something that might be more applicable to most servers and games, and I think some servers would use it.

**Reply by azalty** [reply]
Yea, this feature should come from another plugin

**Comment by iBoonie** [comment]
I don't think it's _that_ niche, when a client gets banned, they get IP banned. If anyone else from the same IP tries to join they wont be able to, so one would think it would be a good idea to atleast kick the people that were on the same IP as the cheater (they will not be able to join on map change anyway)?
